lib/trino-parquet/src/main/java/io/trino/parquet/reader/flat/FlatColumnReader.java

```diff
@@ -259,6 +266,11 @@ private DataPage readPage()
         else if (page instanceof DataPageV2) {
             readFlatPageV2((DataPageV2) page);
         }
+        // When a data page is compressed, the memory used by the decompressed values data
+        // needs to be accounted for separately
+        int dataPageSizeInBytes = pageReader.arePagesCompressed() ? page.getUncompressedSize() : 0;
```

Comment:
Can the buffer be shared between flat readers? If so, how do you deal with it?

Reply:
The only shared memory between the column readers is the one in AbstractParquetDataSource, that has its own accounting logic. We use Slices from there directly only when the data is uncompressed.
That's why the `pageReader.arePagesCompressed()` condition is used to account for additional memory usage only when pages need to be decompressed.

Reply:
Could you also comment why `when !pageReader.arePagesCompressed() => dataPageSizeInBytes == 0`. It's not obvious from the comment